===================================================================
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -20,9 +20,9 @@
 ###############################################################################
 
 module(
     name = "heu",
-    version = "0.6.0.dev20250123",
+    version = "0.6.0.dev20250514",
     compatibility_level = 1,
 )
 
 bazel_dep(name = "yacl", version = "0.4.5b11-nightly-20250514")
